include/envoy/network/connection_handler.h

```diff
@@ -10,6 +10,11 @@
 #include "envoy/ssl/context.h"
 
 namespace Envoy {
+
+namespace Server {
+class Listener;
+}
+// XXX: Move to Server namespace?
```

Comment:
yeah, from an abstraction level in general we try to make server code depend on common code, but not the other way around. Why was this necessary?

Reply:
Listener config is defined in the Server namespace (Server::Listener), and the ConnectionHandler's addListener() now takes it as an argument. I did not want to include server headers from here, so I used a bare class declaration (in the correct namespace) to circumvent the need for the include.

Could also see if the Listener config class could be moved to the common code.

Reply:
Yeah, let's just move Server::Listener (that interface only) into the Network namespace. That interface doesn't depend on anything else in Server. Please do that in a dedicated PR since it will be just code movement. We can review/merge that quickly and then rebase this.

Reply:
Network namespace already has a Listener (the abstract base class, so something needs to be renamed. How about renaming the Server::Listener as Network::ListenerConfig, as it is just that, "A configuration for an individual listener", as the comment says?

Reply:
+1

Reply:
Will do this and the filter directory split tomorrow, hopefully. In the meanwhile I updated this PR so that you may continue the review as you please :-)